src/include/sof/audio/audio_stream.h

```diff
@@ -365,7 +365,7 @@ static inline int audio_stream_set_params(struct audio_stream *buffer,
 	if (!params)
 		return -EINVAL;
 
-	buffer->runtime_stream_params.frame_fmt = params->frame_fmt;
+	buffer->runtime_stream_params.frame_fmt = (enum sof_ipc_frame)params->frame_fmt;
```

Comment:
Good catch by G++, these should be fixed at source though, but sadly I dont think all C compilers allow this
```diff
diff --git a/src/include/ipc/stream.h b/src/include/ipc/stream.h
index c397f34a1..6466b2cbc 100644
--- a/src/include/ipc/stream.h
+++ b/src/include/ipc/stream.h
@@ -75,7 +75,7 @@ struct sof_ipc_stream_params {
        struct sof_ipc_hdr hdr;
        struct sof_ipc_host_buffer buffer;
        uint32_t direction;     /**< enum sof_ipc_stream_direction */
-       uint32_t frame_fmt;     /**< enum sof_ipc_frame */
+       enum sof_ipc_frame frame_fmt:32;        /**< enum sof_ipc_frame */
        uint32_t buffer_fmt;    /**< enum sof_ipc_buffer_format */
        uint32_t rate;
        uint16_t stream_tag;
```